test/test_torch.py

```diff
@@ -2906,6 +2906,413 @@ def test_use_deterministic_algorithms_beta_warning(self, device):
+                        torch.randn(10, device=device),
+                        0,
+                        torch.tensor([3], device=device),
+                        torch.randn(1, device=device)
+                    ],
+                    {}),
+                None, 'index_copy', None
+            ),
+            (
+                'index_put_',
+                torch.Tensor.index_put_,
+                lambda: (
+                    [
+                        torch.randn(10, device=device),
+                        (torch.tensor([0], device=device),),
+                        torch.randn(1, device=device)
+                    ],
+                    dict(accumulate=False)),
+                None, 'index_put_ with accumulate=False', None
+            ),
+            (
+                'put_',
+                torch.Tensor.put_,
+                lambda: (
+                    [
+                        torch.randn(10, device=device),
+                        torch.tensor(0, device=device),
+                        torch.randn(1, device=device)
+                    ],
+                    dict(accumulate=False)),
+                None, 'put_', None
+            ),
+            (
+                'put__accumulate',
+                torch.Tensor.put_,
+                lambda: (
+                    [
+                        torch.randn(10, device=device),
+                        torch.tensor(0, device=device),
+                        torch.randn(1, device=device)
+                    ],
+                    dict(accumulate=True)),
+                'cuda', 'put_', None
+            ),
+            (
+                'index_select_backward',
+                torch.index_select,
+                lambda: (
+                    [
+                        torch.randn(10, device=device, requires_grad=True),
+                        0,
+                        torch.tensor([3], device=device)
+                    ],
+                    {}),
+                'cuda', None, 'index_add_cuda_'
+            ),
+            (
+                'repeat_interleave_backward',
+                torch.repeat_interleave,
+                lambda: ([torch.randn(2, 2, device=device, requires_grad=True), 2], {}),
+                'cuda', None, 'index_add_cuda_'
+            ),
+            (
+                'histc',
+                torch.histc,
+                lambda: ([torch.tensor([], device=device)], dict(min=0, max=3)),
+                'cuda', '_histc_cuda', None
+            ),
+            (
+                'bincount',
+                torch.bincount,
+                lambda: ([torch.tensor([], device=device, dtype=torch.long)], {}),
+                'cuda', '_bincount_cuda', None
+            ),
+            (
+                'kthvalue',
+                torch.kthvalue,
+                lambda: ([torch.randn(10, device=device), 4], {}),
+                'cuda', 'kthvalue CUDA', None
+            ),
+            (
+                'median_indices',
```

Comment:
TODO: Some of the current tests, like `test_median_nondeterministic_alert` lower in this file, test the function, method, and out variants. `test_median_nondeterministic_alert` also tests to make sure that when `indices` is not specified, it does not throw a deterministic error. We need to be able to incorporate these cases here

Reply:
handwriting does make special cases easy to support, and a good "template" or "pattern" showing people how to handwrite these tests is comparable to structuring the pattern in code...